src/utils.ts

```diff
@@ -745,6 +745,15 @@ export function promiseTry<T>(fn: () => T): Promise<T> {
     return new Promise((resolve) => resolve(fn()));
 }
 
+// Creates and awaits all promises, running no more than `chunkSize` at the same time
+export async function chunkPromises<T>(fns: (() => Promise<T>)[], chunkSize: number): Promise<T[]> {
+    const results: T[] = [];
+    for (let i = 0; i < fns.length; i += chunkSize) {
+        results.push(...(await Promise.all(fns.slice(i, i + chunkSize).map(fn => fn()))));
+    }
+    return results;
+}
```

Comment:
I couldn't find any existing logic for chunking promises so I added this. It does feel a little bit clumsy admittedly. Not sure if there's a more elegant way of doing this?

Reply:
it seems fine to me - actually pretty clever for using a for loop in this way. Took me a minute to figure out it'd be fine though 😅